### PR TITLE
Added grep

### PR DESCRIPTION
**Pull request by calebzulawski**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/grep/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  build:
+    - {{ compiler('c') }}
+    - make
+  host:
+    - pcre2
+
+test:
+  commands:
+    - ${CONDA_PREFIX}/bin/grep --version
+    - echo hello world | ${CONDA_PREFIX}/bin/grep hello
```
Comment (hmaarrfk) [inline recipes/grep/meta.yaml:26]:
```suggestion
    - test -x ${PREFIX}/bin/grep
    - if [ $(which grep) != ${PREFIX}/bin/grep ]; then echo 1; fi
    - grep --version
    - echo hello world | grep hello
```

one would use `${PREFIX}` in recipes, and `${CONDA_PREFIX}` during other usage.

**Comment by hmaarrfk** [comment]
@conda-forge/core do we want this package at all? seems like it is more updated than the version i get from ubuntu 22.10 (3.7)

File: recipes/grep/meta.yaml (hunk 1 above)
Comment (calebzulawski) [reply recipes/grep/meta.yaml]: thanks, makes sense!

**Comment by calebzulawski** [comment]
In my particular use case, I am managing the entire environment with conda and am not using system packages (including macOS, where the alternative to acquire GNU grep would be something like homebrew).  I just picked 3.9 because it's the latest--ubuntu 23.04 seems to have 3.8, so I expect 3.9 to be available somewhat soon.  If this version is considered too new, I can use an older version.

**Comment by hmaarrfk** [comment]
thanks Isuru!

**Pull request merged**